qr-scanner/qr_scanner/component.py

```diff
@@ -20,12 +20,15 @@ class QrScanner(NoSSRComponent):
     container_style: rx.Var[Dict[str, str]]
     video_style: rx.Var[Dict[str, str]]
 
-    def get_controlled_triggers(self) -> Dict[str, Var]:
+    def get_event_triggers(self) -> Dict[str, Var]:
         """Dict mapping (event -> expected arguments)."""
-        return super().get_controlled_triggers() | {
-            "on_result": rx.EVENT_ARG,
-            "on_decode": rx.EVENT_ARG,
-            "on_error": BaseVar(name=rx.EVENT_ARG.name + "?.message", type_=str),
+
+        return {
+            **super().get_event_triggers(),
+            "on_result": lambda e0: [e0],
+            "on_decode": lambda e0: [e0],
+            "on_error": lambda e0: [BaseVar(name="_e0" + ".message")],
```

Comment:
```suggestion
            "on_error": lambda e0: [BaseVar(name="_e0?.message")],
```

Reply:
What does the `?` do? 
It worked without it for me.

Reply:
it doesn't deref the prop when the var is undefined or null. basically just avoids a runtime error if this gets called in some circumstances with a missing error.

i'm not sure if it's really necessary anymore, but when i was first prototyping this example app, i think there were some browsers that gave back a bogus error object if you denied camera permission or weren't running on localhost/with TLS.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining